release/rke2/rke2.go

```diff
@@ -0,0 +1,86 @@
+package rke2
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v39/github"
+	ecmHTTP "github.com/rancher/ecm-distro-tools/http"
+	"github.com/rancher/ecm-distro-tools/repository"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	goDevURL           = "https://go.dev/dl/?mode=json"
+	githubApiURL       = "https://api.github.com/"
+	imageBuildBaseRepo = "image-build-base"
+)
+
+type goVersionRecord struct {
+	Version string `json:"version"`
+	Stable  bool   `json:"stable"`
+}
+
+type GithubRelease struct {
+	Name string `json:"name"`
+}
+
+func ImageBuildBaseRelease(ctx context.Context, githubToken string) error {
```

Comment:
Can you pass the github client in here? Will make this function easier to mock/test.